process/factory/shard/vmContainerFactory.go

```diff
@@ -165,7 +165,19 @@ func (vmf *vmContainerFactory) Create() (process.VirtualMachinesContainer, error
 
 // Close closes the vm container factory
 func (vmf *vmContainerFactory) Close() error {
-	return vmf.blockChainHookImpl.Close()
+	err := vmf.container.Close()
+	if err != nil {
+		logVMContainerFactory.Error("cannot close container", "error", err)
+		return err
```

Comment:
I would not return error here just to try to also close the blockchain hook.

Reply:
resolved